minvws/nl-kat-coordination
# Fix pydantic serialization of enum on PATCH of tasks

### Changes

* Added the `use_enum_value=True` for the `Task` pydantic models to ensure the use of the value of enums instead of raw enum values
* Added a `Task` serializer for the server API to serialize correctly, especially when partial updates on `PATCH` endpoints are considered
* For sqlalchemy task update statement added the `task.model_dump(mode="json")` parameter to make sure that the correct values are passed in the `update()` method

### Issue

Closes https://github.com/minvws/nl-kat-coordination/issues/2945

### QA

See the initial issue (https://github.com/minvws/nl-kat-coordination/issues/2945) for the specific exception that was raised. This occurred with normal task creation, so to check if the exception is still raised, is to go through onboarding and create a scan task.
